istioctl/pkg/install/pre-check.go

```diff
@@ -0,0 +1,159 @@
+package install
+
+import (
+	"fmt"
+	"io"
+	"strconv"
+	"strings"
+
+	authorizationapi "k8s.io/api/authorization/v1beta1"
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/version"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/pkg/kubectl/genericclioptions/resource"
+)
+
+const (
+	minK8SVersion = "1.11"
+)
+
+type targetResource struct {
+	namespace string
+	group     string
+	version   string
+	name      string
+}
+
+func installPreCheck(istioNamespaceFlag *string, restClientGetter resource.RESTClientGetter, writer io.Writer) error {
+	fmt.Fprintf(writer, "\n")
+	fmt.Fprintf(writer, "Checking the cluster to make sure it is ready for Istio installation...\n")
+	fmt.Fprintf(writer, "\n")
+	fmt.Fprintf(writer, "kubernetes-api\n")
+	fmt.Fprintf(writer, "-----------------------\n")
+	kubeClient, err := createKubeClient(restClientGetter)
+	if err != nil {
+		return fmt.Errorf("failed to initialize the k8s client: %v", err)
+	}
+	fmt.Fprintf(writer, "can initialize the k8s client\n")
+	v, err := kubeClient.Discovery().ServerVersion()
+	if err != nil {
+		return fmt.Errorf("failed to query the Kubernetes API Server: %v", err)
+	}
+	fmt.Fprintf(writer, "can query the Kubernetes API Server\n")
+
+	fmt.Fprintf(writer, "\n")
+	fmt.Fprintf(writer, "kubernetes-version\n")
+	fmt.Fprintf(writer, "-----------------------\n")
+
+	if !checkKubernetesVersion(v) {
+		return fmt.Errorf("The Kubernetes API version: %v is lower than the minimum version: "+minK8SVersion, v)
+	}
+	fmt.Fprintf(writer, "is running the minimum Kubernetes API version\n")
```

Comment:
This message claims the cluster is running the minimum version even if the cluster *exceeds* the minimum.  A better message "Istio is compatible with Kubernetes {ACTUAL-K8S-VERSION}".

Reply:
done